#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material1 "casterMat"
      coulombFriction [
        0
      ]
    }
  ]
}
Viewpoint {
  orientation 0.479140026511158 -0.685256530709859 -0.5484964194180073 4.220381962770112
  position 0.033211372567043665 2.7340779509127278 0.02944763925600816
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 5 5
  floorAppearance Parquetry {
    type "dark strip"
    colorOverride 0 0 0
  }
}
DEF Cylinder Solid {
  translation -0.58885 0.0250399 -0.188006
  rotation -0.3779979942442246 0.6547109900307159 -0.6545769900327562 -2.418935307179586
  children [
    DEF Cylinder Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.05
        radius 0.025
      }
    }
  ]
  name "solid(5)"
}
DEF Box Solid {
  translation -0.624024 0.025 0.186716
  rotation 0 1 0 1.57079
  children [
    DEF Box Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.05 0.05 0.05
      }
    }
  ]
  boundingObject USE Box
  physics Physics {
  }
}
DEF Cyan Solid {
  translation -0.8 -0.049 -0.2
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.9 0.1 0.92
      }
    }
  ]
  name "solid(1)"
}
DEF Black Solid {
  translation -0.8 -0.049 0.72
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.9 0.1 0.92
      }
    }
  ]
  name "solid(2)"
}
DEF Magenta Solid {
  translation -0.05 -0.049 -0.2
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.6 0.1 0.92
      }
    }
  ]
  name "solid(3)"
}
DEF Yellow Solid {
  translation -0.05 -0.049 0.72
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.6 0.1 0.92
      }
    }
  ]
  name "solid(4)"
}
DEF Mosaic_wall Solid {
  translation -1.26 0 -0.67
  scale 0.001 0.001 0.001
  children [
    Solid {
      children [
        DEF Mosaic_wall Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
            name "DefaultMaterial"
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                1510 0 1850, 910 0 1850, 1510 250 1850, 910 250 1850, 1520 250 370, 1520 250 1860, 1520 0 370, 1520 0 1860, 910 250 1860, 910 250 1850, 910 0 1860, 910 0 1850, 1520 250 1860, 910 250 1860, 1520 0 1860, 910 0 1860, 910 250 1850, 910 250 1860, 1510 250 1850, 1520 250 1860, 1510 250 370, 1520 250 370, 1510 0 370, 1520 0 370, 1510 0 1850, 1520 0 1860, 910 0 1850, 910 0 1860, 1510 250 370, 1520 250 370, 1510 0 370, 1520 0 370, 1510 250 1850, 1510 250 370, 1510 0 1850, 1510 0 370, 460 250 1850, 460 0 1850, 10 250 1850, 10 0 1850, 0 250 1860, 0 250 0, 0 0 1860, 0 0 0, 460 250 1860, 0 250 1860, 460 0 1860, 0 0 1860, 460 250 1850, 460 250 1860, 460 0 1850, 460 0 1860, 1520 250 9.99999, 1520 250 0, 10 250 9.99999, 0 250 0, 10 250 1850, 0 250 1860, 460 250 1850, 460 250 1860, 460 0 1850, 460 0 1860, 10 0 1850, 0 0 1860, 10 0 9.99999, 0 0 0, 1520 0 9.99999, 1520 0 0, 0 250 0, 1520 250 0, 0 0 0, 1520 0 0, 10 250 1850, 10 0 1850, 10 250 9.99999, 10 0 9.99999, 1520 250 0, 1520 250 9.99999, 1520 0 0, 1520 0 9.99999, 1520 250 9.99999, 10 250 9.99999, 1520 0 9.99999, 10 0 9.99999
              ]
            }
            normal Normal {
              vector [
                0 0 -1, 0 0 -1, 0 0 -1, 0 0 -1, 1 0 0, 1 0 0, 1 0 0, 1 0 0, -1 0 1.22465e-16, -1 0 1.22465e-16, -1 0 1.22465e-16, -1 0 1.22465e-16, 0 0 1, 0 0 1, 0 0 1, 0 0 1, 0 1 0, 0 1 0, 0 1 0, 0 1 0, 0 1 0, 0 1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 0 -1, 0 0 -1, 0 0 -1, 0 0 -1, -1 0 0, -1 0 0, -1 0 0, -1 0 0, 0 0 -1, 0 0 -1, 0 0 -1, 0 0 -1, -1 0 0, -1 0 0, -1 0 0, -1 0 0, 0 0 1, 0 0 1, 0 0 1, 0 0 1, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 0 1 0, 0 1 0, 0 1 0, 0 1 0, 0 1 0, 0 1 0, 0 1 0, 0 1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 0 -1, 0 0 -1, 0 0 -1, 0 0 -1, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 0 0 1, 0 0 1, 0 0 1, 0 0 1
              ]
            }
            coordIndex [
              0, 1, 2, -1, 2, 1, 3, -1, 4, 5, 6, -1, 6, 5, 7, -1, 8, 9, 10, -1, 10, 9, 11, -1, 12, 13, 14, -1, 14, 13, 15, -1, 16, 17, 18, -1, 18, 17, 19, -1, 18, 19, 20, -1, 20, 19, 21, -1, 22, 23, 24, -1, 24, 23, 25, -1, 24, 25, 26, -1, 26, 25, 27, -1, 28, 29, 30, -1, 30, 29, 31, -1, 32, 33, 34, -1, 34, 33, 35, -1, 36, 37, 38, -1, 38, 37, 39, -1, 40, 41, 42, -1, 42, 41, 43, -1, 44, 45, 46, -1, 46, 45, 47, -1, 48, 49, 50, -1, 50, 49, 51, -1, 52, 53, 54, -1, 54, 53, 55, -1, 54, 55, 56, -1, 56, 55, 57, -1, 56, 57, 58, -1, 58, 57, 59, -1, 60, 61, 62, -1, 62, 61, 63, -1, 62, 63, 64, -1, 64, 63, 65, -1, 64, 65, 66, -1, 66, 65, 67, -1, 68, 69, 70, -1, 70, 69, 71, -1, 72, 73, 74, -1, 74, 73, 75, -1, 76, 77, 78, -1, 78, 77, 79, -1, 80, 81, 82, -1, 82, 81, 83, -1
            ]
          }
        }
      ]
      name ""
    }
  ]
  name "<STL_BINARY>"
  boundingObject USE Mosaic_wall
}
DEF slider_robot Robot {
  translation 0.189999 0.03 -0.510002
  rotation 0 1 0 -1.5707953071795862
  children [
    Display {
      children [
        Shape {
          appearance Appearance {
            material Material {
              emissiveColor 1 1 1
            }
            texture ImageTexture {
              filtering 0
            }
          }
        }
      ]
      width 128
      height 128
    }
    Camera {
      translation -3.48558e-05 0.0596472 0.101331
      rotation 0.00010405098774656044 -0.9914448832436846 -0.13052598462876425 -3.140015307179586
      children [
        Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.01
            radius 0.005
          }
        }
      ]
      name "cam"
      width 128
      height 128
    }
    DEF cameraPoll Solid {
      translation 0 0.0425 0.09
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.02 0.08 0.01
          }
        }
      ]
      name "solid(5)"
    }
    DEF handle HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0 0 0.1125
      }
      device [
        RotationalMotor {
          name "handleMotor"
        }
        PositionSensor {
          name "handleEncoder"
        }
      ]
      endPoint Solid {
        translation 0 0 0.1125
        rotation 1 0 0 0
        children [
          DEF leftSlider SliderJoint {
            jointParameters JointParameters {
              axis -1 0 0
            }
            device [
              LinearMotor {
                name "leftSlider"
              }
            ]
            endPoint Solid {
              translation 0.0675 -0.01 0.0425
              children [
                DEF leftTouchSensor TouchSensor {
                  translation -0.003 0 0.002
                  children [
                    DEF lefttTouchSensor Shape {
                      appearance PBRAppearance {
                        baseColor 0 0 0
                        roughness 1
                        metalness 0
                      }
                      geometry Box {
                        size 0.001 0.01 0.065
                      }
                    }
                  ]
                  name "leftTouchSensor(1)"
                  boundingObject USE lefttTouchSensor
                  physics Physics {
                  }
                }
                DEF leftSlider Shape {
                  appearance PBRAppearance {
                    baseColor 1 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.005 0.025 0.07
                  }
                }
              ]
              name "solid(2)"
              boundingObject USE leftSlider
              physics Physics {
              }
            }
          }
          DEF rightSlider SliderJoint {
            jointParameters JointParameters {
              axis 1 0 0
            }
            device [
              LinearMotor {
                name "rightSlider"
              }
            ]
            endPoint Solid {
              translation -0.0675 -0.01 0.0425
              children [
                DEF rightTouchSensor TouchSensor {
                  translation 0.003 0 0.002
                  children [
                    DEF rightTouchSensor Shape {
                      appearance PBRAppearance {
                        baseColor 0 0 0
                        roughness 1
                        metalness 0
                      }
                      geometry Box {
                        size 0.001 0.01 0.065
                      }
                    }
                  ]
                  name "rightTouchSensor"
                  boundingObject USE rightTouchSensor
                  physics Physics {
                  }
                }
                DEF rightSlider Shape {
                  appearance PBRAppearance {
                    baseColor 1 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.005 0.025 0.07
                  }
                }
              ]
              name "solid(4)"
              boundingObject USE rightSlider
              physics Physics {
              }
            }
          }
          DEF rightHandle Solid {
            translation -0.045 -0.01 0.005
            children [
              DEF leftHandle Shape {
                appearance PBRAppearance {
                  baseColor 0 0 1
                  roughness 1
                  metalness 0
                }
                geometry Box {
                  size 0.05 0.025 0.005
                }
              }
            ]
            name "solid(1)"
            boundingObject USE leftHandle
            physics Physics {
            }
          }
          DEF leftHandle Solid {
            translation 0.045 -0.01 0.005
            children [
              DEF leftHandle Shape {
                appearance PBRAppearance {
                  baseColor 0 0 1
                  roughness 1
                  metalness 0
                }
                geometry Box {
                  size 0.05 0.025 0.005
                }
              }
            ]
            name "solid(8)"
            boundingObject USE leftHandle
            physics Physics {
            }
          }
          DEF handle Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.14 0.01 0.005
            }
          }
        ]
        name "solid(4)"
        boundingObject USE handle
        physics Physics {
        }
      }
    }
    DEF basePlate Group {
      children [
        Transform {
          translation 0 0 0.005
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0 0 1
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.08 0.005 0.07
              }
            }
          ]
        }
        Transform {
          translation 0 0 0.075
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0 1 1
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.14 0.005 0.07
              }
            }
          ]
        }
      ]
    }
    DEF jointLeft HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.057 0 0
      }
      device [
        RotationalMotor {
          name "leftMotor"
        }
      ]
      endPoint Solid {
        translation 0.057 -1.1083861204024168e-06 2.8748436086387954e-19
        rotation -1.2978932984968527e-13 -1.2968601625741456e-13 -1 1.5699999999999996
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.025
              radius 0.03
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF jointRight HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.057 0 0
      }
      device [
        RotationalMotor {
          name "rightMotor"
        }
      ]
      endPoint Solid {
        translation -0.057 -1.108386120402416e-06 2.8725884028278994e-19
        rotation -1.2968751504934866e-13 1.2958428249805175e-13 1 1.569999999999999
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.025
              radius 0.03
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF casterLeft Solid {
      translation 0.055 -0.01625 0.09
      children [
        DEF caster Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.01375
          }
        }
      ]
      name "solid(2)"
      contactMaterial "casterMat"
      boundingObject USE caster
      physics Physics {
      }
    }
    DEF casterRight Solid {
      translation -0.055 -0.01625 0.09
      rotation 0 0 1 1.57
      children [
        DEF caster Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.01375
          }
        }
      ]
      name "solid(3)"
      contactMaterial "casterMat"
      boundingObject USE caster
      physics Physics {
      }
    }
  ]
  name "robot(1)"
  boundingObject USE basePlate
  physics Physics {
  }
  controller "testing_controller"
}
